### PR TITLE
Logs empty conversion results as errors in a tenant specific standby process

### PR DESCRIPTION
**Pull request by meg1502**
only in case of an error we query the processes framework, as we want to be as quick as possible for successful conversions

Fixes: OX-6843

### CODE REVIEW

**Review by sabieber — APPROVED** [review]

File: src/main/java/sirius/biz/storage/layer2/variants/ConversionEngine.java [hunk 1]
```diff
@@ -170,42 +176,55 @@ public Future performConversion(ConversionProcess conversionProcess) {
...
-                conversionDuration.addValue(conversionProcess.getConversionDuration());
-                result.success();
-            } catch (Exception e) {
-                result.fail(e);
-            }
-        });
+        tasks.executor(EXECUTOR_STORAGE_CONVERSION)
+             .dropOnOverload(() -> result.fail(new IllegalStateException("Conversion subsystem overloaded!")))
+             .fork(() -> {
+                 try {
```
Comment (andyHa) [inline src/main/java/sirius/biz/storage/layer2/variants/ConversionEngine.java:182]: extract method?

File: src/main/java/sirius/biz/storage/layer2/variants/ConversionEngine.java [hunk 2]
```diff
@@ -170,42 +176,55 @@ public Future performConversion(ConversionProcess conversionProcess) {
...
+                                                                                                                  .getVariantName())
+                                                                                             .withContext("filename",
+                                                                                                          conversionProcess
+                                                                                                                  .getBlobToConvert()
+                                                                                                                  .getFilename())));
+                         throw new IllegalArgumentException(Strings.apply(
+                                 "The conversion engine created an empty result for variant %s of %s (%s)",
+                                 conversionProcess.getVariantName(),
+                                 conversionProcess.getBlobToConvert().getFilename(),
+                                 conversionProcess.getBlobToConvert().getBlobKey()));
```
Comment (andyHa) [inline src/main/java/sirius/biz/storage/layer2/variants/ConversionEngine.java:219]: extract method?

**Review by andyHa — CHANGES_REQUESTED** [review]

File: src/main/resources/biz_de.properties [hunk 3]
```diff
@@ -899,3 +899,5 @@ XMLImportJob.invalidXMLDetected = Der Import der Datei '${fileName}' wurde abgeb
 XMLImportJobFactory.xsdSchema = XML-Schema prüfen
 XMLImportJobFactory.xsdSchema.help = Bei Auswahl eines der vorgegebenen XML-Schemata erfolgt der Import nur, wenn die Datei der vorgegebenen Struktur entspricht.
 XMLValidatorErrorHandler.error = Während der Validierung ist ein Fehler aufgetreten in Zeile ${line}: ${message}
+ConversionEngine.processTitle = Bildkonvertierung
```
Comment (andyHa) [inline src/main/resources/biz_de.properties:902]: Könnte auch ein Video etc. sein - vllt "Datei-Konvertierung" oder sowas(?)

Comment (sabieber) [reply src/main/resources/biz_de.properties:902]: Oder evtl "Medienkonvertierung"?

File: src/main/java/sirius/biz/storage/layer2/variants/ConversionEngine.java [hunk 4]
```diff
@@ -178,57 +178,62 @@ public Future performConversion(ConversionProcess conversionProcess) {
...
+        } catch (Exception e) {
+            result.fail(e);
+        }
+    }
+
+    private void handleEmptyResult(ConversionProcess conversionProcess, FileHandle resultFileHandle) {
+        if (resultFileHandle != null) {
+            resultFileHandle.close();
+        }
+        processes.executeInStandbyProcessForCurrentTenant("conversion",
```
Comment (andyHa) [inline src/main/java/sirius/biz/storage/layer2/variants/ConversionEngine.java:216]: is the tenant always properly set (e.g. when BlobDispatcher is in action)? oder do we have to supply it manually

File: src/main/java/sirius/biz/storage/layer2/variants/ConversionEngine.java [hunk 5]
```diff
@@ -178,57 +178,62 @@ public Future performConversion(ConversionProcess conversionProcess) {
...
+                                                          () -> NLS.get("ConversionEngine.processTitle"),
+                                                          processContext -> processContext.log(ProcessLog.error()
```
Comment (andyHa) [inline src/main/java/sirius/biz/storage/layer2/variants/ConversionEngine.java:218]: extract again - this break horribly

**Review by andyHa — CHANGES_REQUESTED** [review]

**Review by andyHa — APPROVED** [review]

**Review by sabieber — APPROVED** [review]

**Review by idlira — APPROVED** [review]

**Review by qw3ry — APPROVED** [review]

**Review by mkeckmkeck — APPROVED** [review]

**Pull request merged**